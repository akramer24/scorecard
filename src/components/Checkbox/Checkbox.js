import React from 'react';
import classNames from 'classnames';
import { omitBy, isUndefined } from 'lodash';
import PropTypes from 'prop-types';
import withForm from '../hocs/withForm';

const Checkbox = ({
  autoFocus,
  checked,
  className,
  disabled,
  id,
  name,
  onChange,
  required,
}) => {
  const passthroughToCheckbox = omitBy({
    autoFocus,
    checked,
    disabled,
    id,
    name,
    onChange,
    required
  }, isUndefined);

  return (
    <input
      className={classNames('ari-ui-checkbox', { [className]: className })}
      type="checkbox"
      {...passthroughToCheckbox}
    />
  )
}

Checkbox.defaultProps = {
  /**
   * Determines whether or not Checkbox has focus on mount.
   */
  autoFocus: PropTypes.bool,
  /**
   * Determines whether or not Checkbox is checked.
   */
  checked: PropTypes.bool,
  /**
   * CSS class(es) applied to Checkbox.
   */
  className: PropTypes.string,
  /**
   * Determines whether or not Checkbox is disabled
   */
  disabled: PropTypes.bool,
  /**
   * CSS id applied to Checkbox.
   */
  id: PropTypes.string,
  /**
   * Name of Checkbox.
   */
  name: PropTypes.string,
  /**
   * Callback griggered on check/uncheck.
   */
  onChange: PropTypes.func,
  /**
   * Determines whether or not Checkbox is required.
   */
  required: PropTypes.bool
}

export default withForm(Checkbox);