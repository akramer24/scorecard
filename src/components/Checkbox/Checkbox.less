.ari-ui-checkbox {
  height: 18px;
  width: 18px;
  appearance: none;
	background-color: #fafafa;
	box-shadow: 0 1px 2px rgba(0,0,0,0.05), inset 0px -15px 10px -12px rgba(0,0,0,0.05);
	padding: 9px;
	border-radius: 3px;
	display: inline-block;
  position: relative;
  .ari-ui-form-field();
  
  &:checked {
    background-color: #e9ecee;
    box-shadow: 0 1px 2px rgba(0,0,0,0.05), inset 0px -15px 10px -12px rgba(0,0,0,0.05), inset 15px 10px -12px rgba(255,255,255,0.1);
    color: #99a1a7;
  }

  &:active, &:checked:active {
    box-shadow: 0 1px 2px rgba(0,0,0,0.05), inset 0px 1px 3px rgba(0,0,0,0.1);
  }

  &:checked:after {
    content: '\2714';
    font-size: 14px;
    position: absolute;
    top: 0px;
    left: 3px;
    color: @primary-color;
  }

  &:focus {
    .ari-ui-text-box-focused();
    .ari-ui-form-field();
  }
}