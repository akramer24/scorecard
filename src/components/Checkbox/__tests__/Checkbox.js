import React from 'react';
import ReactDOM from 'react-dom';
import { configure, mount, render } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import Checkbox from '../Checkbox';

configure({ adapter: new Adapter() });

const mockProps = {
  autoFocus: false,
  checked: false,
  className: 'test-checkbox-class',
  disabled: false,
  id: 'test-checkbox-id',
  name: 'test-checkbox',
  onChange: jest.fn(),
  required: false
}

describe('Checkbox', () => {
  let tree;
  let wrapper;

  beforeAll(() => {
    tree = render(<Checkbox {...mockProps} />);
    wrapper = mount(<Checkbox {...mockProps} />);
  })

  afterAll(() => {
    wrapper.unmount();
  });

  beforeEach(() => {
    jest.clearAllMocks();
    wrapper.setProps(mockProps);
  });

  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Checkbox {...mockProps} />, div);
  });

  it('renders expected Jest snapshot correctly', () => {
    expect(tree).toMatchSnapshot();
  });

  it('renders the data inside it', () => {
    expect(wrapper.text()).toEqual('');
  });

  it('has the CSS class "ari-ui-checkbox"', () => {
    expect(
      wrapper
        .find(`#${mockProps.id}`)
        .last()
        .hasClass('ari-ui-checkbox')
    ).toEqual(true);
  });

  it('calls an onChange callback, if passed via props, when clicked', () => {
    wrapper
      .find(`#${mockProps.id}`)
      .last()
      .simulate('change');
    expect(mockProps.onChange).toHaveBeenCalledTimes(1);
  });
});