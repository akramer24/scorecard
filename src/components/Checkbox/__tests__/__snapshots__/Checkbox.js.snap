// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox renders expected Jest snapshot correctly 1`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "class": "ari-ui-form-item ari-ui-form-item-with-label-top",
    },
    "children": Array [
      Object {
        "attribs": Object {
          "class": "ari-ui-form-item-component",
        },
        "children": Array [
          Object {
            "attribs": Object {
              "class": "ari-ui-checkbox test-checkbox-class",
              "id": "test-checkbox-id",
              "name": "test-checkbox",
              "type": "checkbox",
            },
            "children": Array [],
            "name": "input",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
              "id": undefined,
              "name": undefined,
              "type": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
              "id": undefined,
              "name": undefined,
              "type": undefined,
            },
          },
        ],
        "name": "div",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": [Circular],
        "prev": null,
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
        },
      },
    ],
    "name": "div",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;
